src/Umbraco.Infrastructure/Packaging/PackageDataInstallation.cs

```diff
@@ -546,13 +546,32 @@ public IReadOnlyList<T> ImportDocumentTypes<T>(IReadOnlyCollection<XElement> uns
                 {
                     var alias = documentType.Element("Info").Element("Alias").Value;
                     var folders = foldersAttribute.Value.Split(Constants.CharArrays.ForwardSlash);
+
+
+                    var folderKeysAttribute = documentType.Attribute("FolderKeys");
+
+                    var folderKeys = Array.Empty<Guid>();
+                    if (folderKeysAttribute != null)
+                    {
+                        folderKeys = folderKeysAttribute.Value.Split(Constants.CharArrays.ForwardSlash).Select(x=>Guid.Parse(x)).ToArray();
+                    }
                     var rootFolder = WebUtility.UrlDecode(folders[0]);
-                    //level 1 = root level folders, there can only be one with the same name
-                    var current = _contentTypeService.GetContainers(rootFolder, 1).FirstOrDefault();
+
+                    EntityContainer current;
+                    if(folderKeys.Length == folders.Length)
```

Comment:
If I'm reading it right... looks like there could be a bug here if `folders.Length == 0`.  `folderKeys.Length == folders.Length` would be `true` and then we'd be looking for an element that doesn't exist `folderKeys[0]`.